packages/machine/src/protocol/setup.ts

```diff
@@ -80,13 +80,28 @@ export const SETUP_PROTOCOL: ProtocolExecutionFlow = {
 };
 
 function proposeStateTransition(message: ProtocolMessage, context: Context) {
-  const { multisigAddress } = message.params as SetupParams;
-  const sc = context.stateChannelsMap.get(multisigAddress)!;
-  if (sc === undefined) {
-    console.log("sc keys=", context.stateChannelsMap.keys());
-    throw Error(`no such channel at multisig address ${multisigAddress}`);
+  const {
+    multisigAddress,
+    initiatingAddress,
+    respondingAddress
+  } = message.params as SetupParams;
+
+  if (!context.stateChannelsMap.has(multisigAddress)) {
```

Comment:
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll question for you. Should this function do the logic below this comment? Namely, creating a new `StateChannel` object in the case where a Setup Protocol message is received from a counterparty and it is responding?